### DISCUSSION TITLE
What does "k" mean in http?

### DISCUSSION THREAD

**Discussion by FPNL** [nodejs/help#4934] [Q&A] [answered]
I was reading http, and I found there is k words like below... 

What does "k" mean?

https://github.com/nodejs/node/blob/bfa6e37204f1343c8a340f44c923858833da55b8/lib/_http_incoming.js#L37
```js
const kHeaders = Symbol('kHeaders');
const kHeadersCount = Symbol('kHeadersCount');
const kTrailers = Symbol('kTrailers');
const kTrailersCount = Symbol('kTrailersCount');
```

**Comment by gireeshpunathil** [comment, marked as answer]
it is probably inherited from the variable naming convention from `v8` source - which uses `k` prefix for variable names that are constants. For example: https://github.com/nodejs/node/blob/bfa6e37204f1343c8a340f44c923858833da55b8/deps/v8/include/v8.h#L433

pls refer to [hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation#Examples) for additional context.

/cc @addaleax to see if there is a better explanation.

**Reply by addaleax** [reply]
> which uses k prefix for variable names that are constants.

Yeah – k for konstant. :)

> ls refer to hungarian notation for additional context.

The origins of using `k` this way might be related to hungarian notation, but to avoid confusion, we’re not using hungarian notation in V8 or Node.js, and the most commonly used variant of the hungarian notation (the one used by the Windows API, afaik) uses `c` for constants.

**Reply by FPNL** [reply]
Got it. Thank you

**Comment by jhunterkohler** [comment]
To add more niche knowledge onto the answer: `v8` uses this convention due to the [Google C++ Style](https://google.github.io/styleguide/cppguide.html#Constant_Names).  See the 'Constant Names section.
